src/rust/engine/src/rule_graph.rs

```diff
@@ -744,9 +792,16 @@ pub struct RuleEdges {
   selector_to_dependencies: HashMap<Vec<Selector>, Entries>
 }
 
+impl hash::Hash for RuleEdges {
+  fn hash<H: hash::Hasher>(&self, state: &mut H) {
+    self.dependencies.hash(state);
+    self.selector_to_dependencies.keys().collect::<Vec<_>>().hash(state);
```

Comment:
That would make sense. The other thing I have on my todo list is to reduce the copies of these that are passed around. The only Nodes that really need to have these are Task and Snapshot. Selector nodes only need the entries that match their subselectors. It'd probably be clearer that way too.